presto-main/src/test/java/com/facebook/presto/operator/scalar/TestHyperLogLogFunctions.java

```diff
@@ -0,0 +1,147 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.google.common.io.BaseEncoding;
+import io.airlift.slice.Slice;
+import io.airlift.stats.cardinality.HyperLogLog;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+
+public class TestHyperLogLogFunctions
+        extends AbstractTestFunctions
+{
+    private TestHyperLogLogFunctions() {}
+
+    private static final int NUMBER_OF_BUCKETS = 32768;
+
+    @Test
+    public void testCardinalityNullArray()
+    {
+        String projection = "cardinality(merge_hll(ARRAY[null, null, null]))";
+        assertFunction(projection, BIGINT, null);
+    }
+
+    @Test
+    public void testCardinalityMultipleNullColumns()
+    {
+        String projection = "cardinality(merge_hll(ARRAY[null, null, null]))";
+        assertFunction(projection, BIGINT, null);
+    }
+
+    @Test
+    public void testCardinalityNoColumns()
+    {
+        long blockSize = 0;
+        long uniqueElts = 10000 * blockSize;
+
+        String projection = getCardinalityProjection(getUniqueElements((int) blockSize, uniqueElts));
+        assertFunction(projection, BIGINT, null);
+    }
+
+    @Test
+    public void testCardinalitySingleColumn()
+    {
+        long blockSize = 1;
+        long uniqueElts = 10000 * blockSize;
+
+        String projection = getCardinalityProjection(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        functionAssertions.assertFunctionWithError(projection, BIGINT, uniqueElts, error);
+    }
+
+    @Test
+    public void testCardinalityTwoColumns()
+    {
+        long blockSize = 2;
+        long uniqueElts = 10000 * blockSize;
+
+        String projection = getCardinalityProjection(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        functionAssertions.assertFunctionWithError(projection, BIGINT, uniqueElts, error);
+    }
+
+    @Test
+    public void testCardinalityThreeColumns()
+    {
+        long blockSize = 3;
+        long uniqueElts = 10000 * blockSize;
+
+        String projection = getCardinalityProjection(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        functionAssertions.assertFunctionWithError(projection, BIGINT, uniqueElts, error);
+    }
+
+    @Test
+    public void testCardinalityManyColumns()
+    {
+        // max number of columns to merge is 254
+        long blockSize = 254;
+        long uniqueElts = 1000 * blockSize;
+        String projection = getCardinalityProjection(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        functionAssertions.assertFunctionWithError(projection, BIGINT, uniqueElts, error);
+    }
+
+    private HyperLogLog[] getUniqueElements(int blockSize, long uniqueElements)
```

Comment:
For the varargs I think it's OK and we can leave it.  Varargs are good for readability.

For the output, lets use

```java
ImmutableList.Builder<HyperLogLog> builder = ImmutableList.builder();
build.add(...) // however many times needed
return builder.build;
```